Sources/Plasma/FeatureLib/pfPasswordStore/CMakeLists.txt

```diff
@@ -8,36 +8,32 @@ set(pfPasswordStore_SOURCES
 )
 
 if(WIN32 AND NOT CYGWIN)
-    set(pfPasswordStore_SOURCES ${pfPasswordStore_SOURCES}
-        pfPasswordStore_Win.cpp
-    )
-endif(WIN32 AND NOT CYGWIN)
-
-if(HAVE_LIBSECRET)
-    add_definitions(-DHAVE_LIBSECRET)
-    set(pfPasswordStore_SOURCES ${pfPasswordStore_SOURCES}
-        pfPasswordStore_Unix.cpp
-    )
-    include_directories(${LIBSECRET_INCLUDE_DIRS})
-endif(HAVE_LIBSECRET)
-
-if(APPLE)
-    set(pfPasswordStore_SOURCES ${pfPasswordStore_SOURCES}
-        pfPasswordStore_Apple.cpp
-    )
-endif(APPLE)
+    list(APPEND pfPasswordStore_SOURCES pfPasswordStore_Win.cpp)
+elseif(TARGET PkgConfig::LIBSECRET)
+    list(APPEND pfPasswordStore_SOURCES pfPasswordStore_Unix.cpp)
+elseif(TARGET Security::Security)
+    list(APPEND pfPasswordStore_SOURCES pfPasswordStore_Apple.cpp)
```

Comment:
We should probably put the Security case above the libsecret case, since Security is guaranteed to exist on macOS and we want to use it, but someone hypothetically could also compile libsecret for macOS.

Reply:
I realize the solution may look weird, but the cmake documentation implies that multiple `elseif()` blocks may execute.